### PR TITLE
[ECP-9498] Implement `onError` callback for the Paypal.

### PR DESCRIPTION
**Pull request by khushboo-singhvi**
**Description**
While abandoning the payment from Paypal, the order is not cancelled, it remains in Pending Payment state until we have the OFFER_CLOSED processed, blocking the inventory. This PR aims to implement `onError` event being triggered for Paypal cancelation events.

**Tested scenarios**
- Canceling or abondoning Paypal orders without completing them
- E2E automated test case has been implemented on [#135](https://github.com/Adyen/adyen-integration-tools-tests/pull/135)

Fixes  
#2748

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js [hunk 1]
```diff
@@ -374,15 +375,21 @@ define(
 
                 try {
                     const orderId = await placeOrderAction(data, self.currentMessageContainer);
-                    self.afterPlaceOrder();
+                    //self.afterPlaceOrder();
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:378]:
```suggestion
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 2]
```diff
@@ -32,14 +36,45 @@ define(
...
+                   this.token = action.sdkData.token;
                 this.actionComponent = component.handleAction(action);
             },
             handleOnFailure: function(response, component) {
                 this.isPlaceOrderAllowed(true);
                 fullScreenLoader.stopLoader();
                 component.handleReject(response);
             },
+            handleOnError:  function (error, component) {
+                console.log("onError:",error);
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:48]: Can we safely log any error message coming from the component? What would you think about logging to console only if the environment is TEST ([example](https://github.com/Adyen/adyen-magento2/blob/024fe63d83e6d7360bee290540df1d86652ef738/view/frontend/web/js/model/adyen-checkout.js#L65))?

Comment (khushboo-singhvi) [reply view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:48]: Oh, left from my testing. Updating it. Thank You!

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 3]
```diff
@@ -32,14 +36,45 @@ define(
...
+                if (!!component.paymentData) {
+                    request.paymentData = component.paymentData;
+                }
+                //Create details array for the payload
+                let details ={};
+                if(!!this.token) {
+                    details.orderID= this.token;
+                }
+                request.details = details;
+                request.cancelled = true;
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:61]: Where does the plugin consume `cancelled: true` argument?

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 4]
```diff
@@ -12,11 +12,15 @@ define(
...
     function(
         quote,
         adyenPaymentMethod,
-        fullScreenLoader
+        fullScreenLoader,
+        adyenPaymentService,
+        errorProcessor
     ) {
         return adyenPaymentMethod.extend({
             placeOrderButtonVisible: false,
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:26]: To be on the safe side, let's define `token` as a property with default value `null`.

```suggestion
            placeOrderButtonVisible: false,
            token: null,
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 5]
```diff
@@ -32,14 +36,45 @@ define(
...
+            handleOnError:  function (error, component) {
+                console.log("onError:",error);
+
+                // call endpoint with component.paymentData if available
+                let request = {};
+                if (!!component.paymentData) {
+                    request.paymentData = component.paymentData;
+                }
+                //Create details array for the payload
+                let details ={};
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:59]:
```suggestion
                let details = {};
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 6]
```diff
@@ -32,14 +36,45 @@ define(
...
+                if(!!this.token) {
+                    details.orderID= this.token;
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:61]:
```suggestion
                    details.orderID = this.token;
```

**Review by candemiralp — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Pull request merged**